File: Vanilla/sources/zdoc/media/lua/shared/Library/Stats.lua
---@class Stats : zombie.characters.Stats
---@field public Anger float
---@field public boredom float
---@field public endurance float
---@field public enduranceRecharging boolean
---@field public endurancelast float
---@field public endurancedanger float
---@field public endurancewarn float
---@field public fatigue float
---@field public fitness float
---@field public hunger float
---@field public idleboredom float
---@field public morale float
---@field public stress float
---@field public Fear float
---@field public Panic float
---@field public Sanity float
---@field public Sickness float
---@field public Boredom float
---@field public Pain float
---@field public Drunkenness float
---@field public NumVisibleZombies int
---@field public LastNumVisibleZombies int
---@field public Tripping boolean
---@field public TrippingRotAngle float
---@field public thirst float
---@field public NumChasingZombies int
---@field public LastVeryCloseZombies int
---@field public NumCloseZombies int
---@field public LastNumChasingZombies int
---@field public stressFromCigarettes float
---@field public ChasingZombiesDanger float
---@field public MusicZombiesVisible int
---@field public MusicZombiesTargeting int
Stats = {}

---@public
---@return float @the stress
function Stats:getStress() end

---@public
---@return float @the endurancelast
function Stats:getEndurancelast() end

---throws java.io.IOException
---@public
---@param input DataInputStream
---@return void
---@overload fun(input:ByteBuffer, WorldVersion:int)
function Stats:load(input) end

---throws java.io.IOException
---@public
---@param input ByteBuffer
---@param WorldVersion int
---@return void
function Stats:load(input, WorldVersion) end

---@public
---@param Panic float @the Panic to set
---@return void
function Stats:setPanic(Panic) end

---@public
---@return float @the hunger
function Stats:getHunger() end

---@public
---@param Fear float @the Fear to set
---@return void
function Stats:setFear(Fear) end

---@public
---@param fitness float @the fitness to set
---@return void
function Stats:setFitness(fitness) end

---@public
---@param endurance float @the endurance to set
---@return void
function Stats:setEndurance(endurance) end

---@public
---@param arg0 float
---@return void
function Stats:setStressFromCigarettes(arg0) end

---@public
---@return boolean
function Stats:getEnduranceRecharging() end

---@public
---@return float @the Sanity
function Stats:getSanity() end

---@public
---@return float @the Sickness
function Stats:getSickness() end

---@public
---@param morale float @the morale to set
---@return void
function Stats:setMorale(morale) end

---@public
---@param boredom float @the boredom to set
---@return void
function Stats:setBoredom(boredom) end

---@public
---@param hunger float @the hunger to set
---@return void
function Stats:setHunger(hunger) end

---throws java.io.IOException
---@public
---@param output ByteBuffer
---@return void
---@overload fun(output:DataOutputStream)
function Stats:save(output) end

---throws java.io.IOException
---@public
---@param output DataOutputStream
---@return void
function Stats:save(output) end

---@public
---@return int
function Stats:getNumVisibleZombies() end

---@public
---@param NumVisibleZombies int @the NumVisibleZombies to set
---@return void
function Stats:setNumVisibleZombies(NumVisibleZombies) end

---@public
---@param endurancelast float @the endurancelast to set
---@return void
function Stats:setEndurancelast(endurancelast) end

---@public
---@return float @the endurance
function Stats:getEndurance() end

---@public
---@return float @the idleboredom
function Stats:getIdleboredom() end

---@public
---@return float @the Pain
function Stats:getPain() end

---@public
---@return float @the TrippingRotAngle
function Stats:getTrippingRotAngle() end

---@public
---@param Pain float @the Pain to set
---@return void
function Stats:setPain(Pain) end

---@public
---@param Anger float @the Anger to set
---@return void
function Stats:setAnger(Anger) end

---@public
---@param fatigue float @the fatigue to set
---@return void
function Stats:setFatigue(fatigue) end

---@public
---@return void
function Stats:resetStats() end

---@public
---@return float @the Fear
function Stats:getFear() end

---@public
---@param Sanity float @the Sanity to set
---@return void
function Stats:setSanity(Sanity) end

---@public
---@param Tripping boolean @the Tripping to set
---@return void
function Stats:setTripping(Tripping) end

---@public
---@return float @the endurancedanger
function Stats:getEndurancedanger() end

---@public
---@return int @the NumVisibleZombies
function Stats:getVisibleZombies() end

---@public
---@return float
function Stats:getStressFromCigarettes() end

---@public
---@return float @the fitness
function Stats:getFitness() end

---@public
---@return float @the morale
function Stats:getMorale() end

---@public
---@return float @the boredom
function Stats:getBoredom() end

---@public
---@return float @the Anger
function Stats:getAnger() end

---@public
---@return int
function Stats:getNumChasingZombies() end

---@public
---@param endurancedanger float @the endurancedanger to set
---@return void
function Stats:setEndurancedanger(endurancedanger) end

---@public
---@return boolean @the Tripping
function Stats:isTripping() end

---@public
---@return float @the Panic
function Stats:getPanic() end

---@public
---@param idleboredom float @the idleboredom to set
---@return void
function Stats:setIdleboredom(idleboredom) end

---@public
---@return float @the thirst
function Stats:getThirst() end

---@public
---@param endurancewarn float @the endurancewarn to set
---@return void
function Stats:setEndurancewarn(endurancewarn) end

---@public
---@param thirst float @the thirst to set
---@return void
function Stats:setThirst(thirst) end

---@public
---@return float @the fatigue
function Stats:getFatigue() end

---@public
---@param Sickness float @the Sickness to set
---@return void
function Stats:setSickness(Sickness) end

---@public
---@param Drunkenness float @the Drunkenness to set
---@return void
function Stats:setDrunkenness(Drunkenness) end

---@public
---@return float
function Stats:getMaxStressFromCigarettes() end

---@public
---@param TrippingRotAngle float @the TrippingRotAngle to set
---@return void
function Stats:setTrippingRotAngle(TrippingRotAngle) end

---@public
---@return float @the Drunkenness
function Stats:getDrunkenness() end

---@public
---@return float @the endurancewarn
function Stats:getEndurancewarn() end

---@public
---@param stress float @the stress to set
---@return void
function Stats:setStress(stress) end

File: Vanilla/sources/zomboid/zombie/characters/Stats.java
package zombie.characters;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.nio.ByteBuffer;
import zombie.core.math.PZMath;

public final class Stats {
   public float Anger = 0.0F;
   public float boredom = 0.0F;
   public float endurance = 1.0F;
   public boolean enduranceRecharging = false;
   public float endurancelast = 1.0F;
   public float endurancedanger = 0.25F;
   public float endurancewarn = 0.5F;
   public float fatigue = 0.0F;
   public float fitness = 1.0F;
   public float hunger = 0.0F;
   public float idleboredom = 0.0F;
   public float morale = 0.5F;
   public float stress = 0.0F;
   public float Fear = 0.0F;
   public float Panic = 0.0F;
   public float Sanity = 1.0F;
   public float Sickness = 0.0F;
   public float Boredom = 0.0F;
   public float Pain = 0.0F;
   public float Drunkenness = 0.0F;
   public int NumVisibleZombies = 0;
   public int LastNumVisibleZombies = 0;
   public boolean Tripping = false;
   public float TrippingRotAngle = 0.0F;
   public float thirst = 0.0F;
   public int NumChasingZombies = 0;
   public int LastVeryCloseZombies = 0;
   public static int NumCloseZombies = 0;
   public int LastNumChasingZombies = 0;
   public float stressFromCigarettes = 0.0F;
   public float ChasingZombiesDanger;
   public int MusicZombiesVisible = 0;
   public int MusicZombiesTargeting = 0;

   public int getNumVisibleZombies() {
      return this.NumVisibleZombies;
   }

   public int getNumChasingZombies() {
      return this.LastNumChasingZombies;
   }

   public void load(DataInputStream var1) throws IOException {
      this.Anger = var1.readFloat();
      this.boredom = var1.readFloat();
      this.endurance = var1.readFloat();
      this.fatigue = var1.readFloat();
      this.fitness = var1.readFloat();
      this.hunger = var1.readFloat();
      this.morale = var1.readFloat();
      this.stress = var1.readFloat();
      this.Fear = var1.readFloat();
      this.Panic = var1.readFloat();
      this.Sanity = var1.readFloat();
      this.Sickness = var1.readFloat();
      this.Boredom = var1.readFloat();
      this.Pain = var1.readFloat();
      this.Drunkenness = var1.readFloat();
      this.thirst = var1.readFloat();
   }

   public void load(ByteBuffer var1, int var2) throws IOException {
      this.Anger = var1.getFloat();
      this.boredom = var1.getFloat();
      this.endurance = var1.getFloat();
      this.fatigue = var1.getFloat();
      this.fitness = var1.getFloat();
      this.hunger = var1.getFloat();
      this.morale = var1.getFloat();
      this.stress = var1.getFloat();
      this.Fear = var1.getFloat();
      this.Panic = var1.getFloat();
      this.Sanity = var1.getFloat();
      this.Sickness = var1.getFloat();
      this.Boredom = var1.getFloat();
      this.Pain = var1.getFloat();
      this.Drunkenness = var1.getFloat();
      this.thirst = var1.getFloat();
      if (var2 >= 97) {
         this.stressFromCigarettes = var1.getFloat();
      }

   }

   public void save(DataOutputStream var1) throws IOException {
      var1.writeFloat(this.Anger);
      var1.writeFloat(this.boredom);
      var1.writeFloat(this.endurance);
      var1.writeFloat(this.fatigue);
      var1.writeFloat(this.fitness);
      var1.writeFloat(this.hunger);
      var1.writeFloat(this.morale);
      var1.writeFloat(this.stress);
      var1.writeFloat(this.Fear);
      var1.writeFloat(this.Panic);
      var1.writeFloat(this.Sanity);
      var1.writeFloat(this.Sickness);
      var1.writeFloat(this.Boredom);
      var1.writeFloat(this.Pain);
      var1.writeFloat(this.Drunkenness);
      var1.writeFloat(this.thirst);
   }

   public void save(ByteBuffer var1) throws IOException {
      var1.putFloat(this.Anger);
      var1.putFloat(this.boredom);
      var1.putFloat(this.endurance);
      var1.putFloat(this.fatigue);
      var1.putFloat(this.fitness);
      var1.putFloat(this.hunger);
      var1.putFloat(this.morale);
      var1.putFloat(this.stress);
      var1.putFloat(this.Fear);
      var1.putFloat(this.Panic);
      var1.putFloat(this.Sanity);
      var1.putFloat(this.Sickness);
      var1.putFloat(this.Boredom);
      var1.putFloat(this.Pain);
      var1.putFloat(this.Drunkenness);
      var1.putFloat(this.thirst);
      var1.putFloat(this.stressFromCigarettes);
   }

   public float getAnger() {
      return this.Anger;
   }

   public void setAnger(float var1) {
      this.Anger = var1;
   }

   public float getBoredom() {
      return this.boredom;
   }

   public void setBoredom(float var1) {
      this.boredom = var1;
   }

   public float getEndurance() {
      return this.endurance;
   }

   public void setEndurance(float var1) {
      this.endurance = var1;
   }

   public float getEndurancelast() {
      return this.endurancelast;
   }

   public void setEndurancelast(float var1) {
      this.endurancelast = var1;
   }

   public float getEndurancedanger() {
      return this.endurancedanger;
   }

   public void setEndurancedanger(float var1) {
      this.endurancedanger = var1;
   }

   public float getEndurancewarn() {
      return this.endurancewarn;
   }

   public void setEndurancewarn(float var1) {
      this.endurancewarn = var1;
   }

   public boolean getEnduranceRecharging() {
      return this.enduranceRecharging;
   }

   public float getFatigue() {
      return this.fatigue;
   }

   public void setFatigue(float var1) {
      this.fatigue = var1;
   }

   public float getFitness() {
      return this.fitness;
   }

   public void setFitness(float var1) {
      this.fitness = var1;
   }

   public float getHunger() {
      return this.hunger;
   }

   public void setHunger(float var1) {
      this.hunger = var1;
   }

   public float getIdleboredom() {
      return this.idleboredom;
   }

   public void setIdleboredom(float var1) {
      this.idleboredom = var1;
   }

   public float getMorale() {
      return this.morale;
   }

   public void setMorale(float var1) {
      this.morale = var1;
   }

   public float getStress() {
      return this.stress + this.getStressFromCigarettes();
   }

   public void setStress(float var1) {
      this.stress = var1;
   }

   public float getStressFromCigarettes() {
      return this.stressFromCigarettes;
   }

   public void setStressFromCigarettes(float var1) {
      this.stressFromCigarettes = PZMath.clamp(var1, 0.0F, this.getMaxStressFromCigarettes());
   }

   public float getMaxStressFromCigarettes() {
      return 0.51F;
   }

   public float getFear() {
      return this.Fear;
   }

   public void setFear(float var1) {
      this.Fear = var1;
   }

   public float getPanic() {
      return this.Panic;
   }

   public void setPanic(float var1) {
      this.Panic = var1;
   }

   public float getSanity() {
      return this.Sanity;
   }

   public void setSanity(float var1) {
      this.Sanity = var1;
   }

   public float getSickness() {
      return this.Sickness;
   }

   public void setSickness(float var1) {
      this.Sickness = var1;
   }

   public float getPain() {
      return this.Pain;
   }

   public void setPain(float var1) {
      this.Pain = var1;
   }

   public float getDrunkenness() {
      return this.Drunkenness;
   }

   public void setDrunkenness(float var1) {
      this.Drunkenness = var1;
   }

   public int getVisibleZombies() {
      return this.NumVisibleZombies;
   }

   public void setNumVisibleZombies(int var1) {
      this.NumVisibleZombies = var1;
   }

   public boolean isTripping() {
      return this.Tripping;
   }

   public void setTripping(boolean var1) {
      this.Tripping = var1;
   }

   public float getTrippingRotAngle() {
      return this.TrippingRotAngle;
   }

   public void setTrippingRotAngle(float var1) {
      this.TrippingRotAngle = var1;
   }

   public float getThirst() {
      return this.thirst;
   }

   public void setThirst(float var1) {
      this.thirst = var1;
   }

   public void resetStats() {
      this.Anger = 0.0F;
      this.boredom = 0.0F;
      this.fatigue = 0.0F;
      this.hunger = 0.0F;
      this.idleboredom = 0.0F;
      this.morale = 0.5F;
      this.stress = 0.0F;
      this.Fear = 0.0F;
      this.Panic = 0.0F;
      this.Sanity = 1.0F;
      this.Sickness = 0.0F;
      this.Boredom = 0.0F;
      this.Pain = 0.0F;
      this.Drunkenness = 0.0F;
      this.thirst = 0.0F;
   }
}
